docs/website/docs/developers/debugging/gpu.md

```diff
@@ -0,0 +1,234 @@
+
+Though most of the time, we can leverage existing facilities to avoid going down
+the full top-down hiearchical debugging procedure.
+For example, for regression happening on an existing model, CI or `git bitsect`
+might tell us directly the culprit commit.
+
+### .. using tools
+
+For issues with strong signals like crashing, it's also easier to pinpoint the
+exact culprit with dedicated tools--we can leverage various
+[sanitizers](./sanitizers.md) or debuggers.
+
+## Isolating the problematic component
+
+If we are facing a large problem without a clear clue, we need to isolate the
+problematic compiler or runtime layer first, typically by comparing with a
+working solution:
+
+!!! tip "[:material-check-bold:/:material-trending-up:]"
+
+    Sanitize the environment first.
+    Asking these questions and making sure the environment is proper can save
+    you hours of debugging sometimes:
+
+    * Did you recently updated the GPU SDK or driver?
+    * Are others able to reproduce the issue?
+    * If not what SDK / driver versions they are using?
+    * Is your machine drawing enough power when benchmarking?
+    * Is your machine connected with a mointor (e.g., for Vulkan)?
+    * How long since you last rebooted your machine? 👻
+
+!!! tip "[:material-check-bold:/:material-trending-up:]"
+
+    We have multiple GPU targets/drivers in IREE--LLVMGPU/CUDA, LLVMGPU/HIP,
+    SPIR-V/Vulkan, SPIR-V/Metal.
+
+    For the _same_ GPU, we typically have two paths to target, e.g., CUDA/HIP
+    or Vulkan for NVIDIA/AMD GPUs, Metal or Vulkan for Apple GPUs.
+
+    If one path is correct/performant, we can diff against it to try isolate
+    the problem--the common/shared compiler/runtime code is likely okay; what
+    differs between paths is likely problematic.
+
+!!! tip "[:material-check-bold:/:material-trending-up:] [:simple-vulkan:]"
+
+    Vulkan supports different GPUs.
+    Similarly, if one GPU gives correct/performant result, we diff against it
+    to find clues.
+
+    Even more code in compiler/runtime are shared here; what's problematic is
+    likely different capabilities triggering different CodeGen pipelines so
+    revealing bugs in a particular CodeGen pipeline.
+    Or there are driver issues from a particular vendor.
+
+!!! tip "[:material-check-bold:]"
+
+    If the CPU is working properly, we can use the same dispatch region formation
+    and diff against the CPU dispatches one by one to isolate the problem. See
+    [this issue](https://github.com/openxla/iree/issues/14739) as an example.
+
+!!! tip "[:material-check-bold:]"
+
+    `--iree-flow-trace-dispatch-tensors` and/or `--iree-flow-break-dispatch=` to
+    `iree-compile` is quite helpful to inspect the output after all/each
+    dispatch(es).
+
+!!! tip "[:material-check-bold:]"
+
+    `iree-reduce` is a great tool to reduce and isolate issues programmatically.
+    See more details [here](https://github.com/openxla/iree/blob/main/samples/reducer/README.md).
```

Comment:
lol maybe too many admonitions here :P

color is nice, but this is tricky to read

![image](https://github.com/openxla/iree/assets/4010439/01270694-bf27-4ebe-93e9-a2add0df5c47)

Reply:
SGTM